apps/bbsync/query.py

```diff
@@ -201,6 +202,21 @@ def restore_original(self):
         self._json = json.loads(srtnotes) if srtnotes else {}
         self._original_keys = list(self._json.keys())
 
+    def validate(self):
+        """
+        validation safeguard to ensure that we always create valid SRT notes JSON data
+        throws SRTNotesValidationError exception in the case of invalid SRT notes JSON
+        """
+        with open(SRTNOTES_SCHEMA_PATH) as schema_fp:
+            self.srtnotes_schema = json.load(schema_fp)
+
+        try:
+            jsonschema.validate(self._json, schema=self.srtnotes_schema)
+        except jsonschema.ValidationError as e:
+            raise SRTNotesValidationError(
+                f"Invalid JSON produced for SRT notes: {e}"
+            ) from e
```

Comment:
exception chaining is automatic in Python 3, i.e. you don't need to do the `from e` nor embed it into the exception string